### PR TITLE
keep decoys

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

Please include a summary of the change and which issue is fixed.

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### CODE REVIEW

**Comment by timosachsenberg** [comment]
checked:
- feature finder skips decoys: check
- mztab export writes decoys: ?

**Comment by timosachsenberg** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
@jpfeuffer I would probably need some help for the last step of copying the decoys +/- non-quantified ids back into the consensusmap (and how this plays with the filtering of groups etc.).

**Comment by jpfeuffer** [comment]
What do you mean with "copy back"?
Wouldn't it be better if you just load them from the beginning and just ignore them during alignment/linking?

**Comment by timosachsenberg** [comment]
If that is possible it would make a lot of things simpler. Requires more memory though.

**Review by jpfeuffer — APPROVED** [review]
Looks good.

File: src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp [hunk 1]
```diff
@@ -472,9 +472,10 @@ namespace OpenMS
     }
 
     // add back peptide identification id if it got skipped (decoys)
-    for (auto p : peptides)
+    for (const auto& pep : peptides)
```
Comment (jpfeuffer) [inline src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp:475]: You can also flag the copy-for-loop check. Not sure if it makes a difference here now though.

**Pull request merged**